StreetHawk/Classes/Core/Publish/SHBaseViewController.m

```diff
@@ -270,7 +270,13 @@ - (void)presentOnTopWithCover:(BOOL)needCover withCoverColor:(UIColor *)coverCol
             self.coverView.overlayColor = coverColor;
             self.coverView.overlayAlpha = coverAlpha;
             self.coverView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight; //make cover always full screen during rotation.
-            [rootVC.view addSubview:self.coverView];
+            UIViewController *presentedVC = rootVC;
+            while (presentedVC.presentedViewController != nil)
+            {
+                presentedVC = presentedVC.presentedViewController;
+            }
+            [presentedVC.view addSubview:self.coverView];
+//            [[UIApplication sharedApplication].keyWindow addSubview:self.coverView]; //this also work but cannot rotate
```

Comment:
Not. It's a thing I need to research and try later. Too busy to stop here at this stage.